test_utilities/src/common.rs

```diff
@@ -660,3 +667,64 @@ pub fn environment_only_http_server_router(context: &Context, organization_id: &
         clone_from_environment_id: None,
     }
 }
+
+pub fn deploy_upgrade_destroy_infra_and_env<'a>(
+    tx: &'a mut Transaction<'a>,
+    kubernetes: &'a dyn Kubernetes,
+    environment_action: &'a EnvironmentAction,
+) {
+    // Deploy infrastructure
+    if let Err(err) = tx.create_kubernetes(kubernetes) {
```

Comment:
With the transaction commit behavior, this func will do a lot of useless stuff as we talked yesterday

Reply:
Done :)